cordova-plugin-outline/android/java/org/outline/OutlinePlugin.java

```diff
@@ -254,6 +261,28 @@ private void executeAsync(
           boolean isReachable =
               ShadowsocksConnectivity.isServerReachable(args.getString(0), args.getInt(1));
           callback.sendPluginResult(new PluginResult(PluginResult.Status.OK, isReachable));
+        } else if (Action.FETCH_HTTPS.is(action)) {
+          try {
+            Https.Request request = Https.jsonObjectToRequest(args.getJSONObject(0));
+            Https.Response response = Https.fetch(request);
+            callback.success(Https.responseToJsonObject(response));
+          } catch (Exception e) {
+            LOG.log(Level.SEVERE, "failed to fetch HTTPS", e);
+            ErrorCode errorCode = ErrorCode.UNEXPECTED;
```

Comment:
I think we should simplify our error handling.

I was imagining a very simple UI that says "Failed to fetch config (MalformedURLException)" or "Failed to fetch config (ECONNREFUSED)"

All you should need to pass from the native and Electron code to the webview code is a generic FetchConfigError and a string.

For Java, you could return the name of the error. For Node we could return the code.
This simplifies the error handling, and also captures the "unexpected" cases, which are the ones we probably want to find out more about.